From 680a8401aa030878cd4927dbba37a652a24655c6 Mon Sep 17 00:00:00 2001
From: Bai Ping <b51503@freescale.com>
Date: Mon, 15 Dec 2014 23:56:09 +0800
Subject: [PATCH] MLK-9996 arm: imx6: Correct the AHB clock in
 low_bus_freq_mode

When the busfreq is in audio_bus_freq_mode, the AHB bus is at 8MHz,
in low_bus_freq_mode, the AHB needs to run at 24MHz. So when switching
from audio_bus_freq_mode to low_bus_freq_mode, make sure the AHB is at
24MHz in low_bus_freq_mode.

Signed-off-by: Bai Ping <b51503@freescale.com>
---
 arch/arm/mach-imx/busfreq-imx6.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm/mach-imx/busfreq-imx6.c b/arch/arm/mach-imx/busfreq-imx6.c
index 49c2c71..8ffdfab 100644
--- a/arch/arm/mach-imx/busfreq-imx6.c
+++ b/arch/arm/mach-imx/busfreq-imx6.c
@@ -288,6 +288,12 @@ static void enter_lpm_imx6sl(void)
 		} else {
 			if (!ultra_low_bus_freq_mode && !low_bus_freq_mode) {
 				/*
+				 * Anyway, make sure the AHB is running at 24MHz
+				 * in low_bus_freq_mode.
+				 */
+				if (audio_bus_freq_mode)
+					imx_clk_set_rate(ahb_clk, LPAPM_CLK);
+				/*
 				 * Set DDR to 24MHz.
 				 * Since we are going to bypass PLL2,
 				 * we need to move ARM clk off PLL2_PFD2
-- 
1.8.0

